.github/actions/build/action.yml

```diff
@@ -0,0 +1,36 @@
+name: Install build prerequisites
+
+inputs:
+  cache-dir:  
+    description: Where to put vcpkg cache
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Capture vcpkg revision for use in cache key
+      shell: bash
+      run: |
+         git -C vcpkg rev-parse HEAD > vcpkg_commit.txt
+
+    - name: Restore cache
+      uses: actions/cache@v3
+      with:
+        path: ${{ inputs.cache-dir }}
+        key: v01-vcpkg-${{ runner.os }}-${{ hashFiles('vcpkg_commit.txt', 'alternatives/*/vcpkg.json') }}
+        restore-keys: |
+          v01-vcpkg-${{ runner.os }}
```

Comment:
This will probably need some more work.  

1. I personally do not favor the runner.os because it is not specific enough to provide proper vcpkg caching.  We can solve this when we observe the failure, but I believe there will be cases where the windows-2022 and windows-2019 might not be able to share the same cache. (VCPKG has its own caching strategy).  Again, let's not sweat this until we see it.
2. We probably need to solve this before merge... Since the first writer will win in creating the cache, we will need to add the crypto library as a variable in the the cache key.  Are you seeing this fail?  If not, I am surprised.

Reply:
Yeah, good point.  I think we are seeing the crypto libraries get rebuilt, thus causing unnecessary slowness.  Two alternatives come to mind:

1. Including the crypto library in the cache key (as you suggest)
2. Somehow filling all three libraries into the cache

The latter approach seems appealing to me, because you would get more sharing across builds for the same platform.  But it would require some refactoring in how we select the crypto library in CMakeLists.txt.

So in the short run, I think the right fix is (1).  I can make `matrix.os` and `matrix.crypto` inputs to this job.